### ISSUE TITLE
Add tf.normalize to provide a clean interface for tensor normalization

### ISSUE THREAD

**Issue by sleighsoft** [tensorflow/tensorflow#28741]
**System information**
- TensorFlow version (you are using): 1.3+ or 2.0a
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
Currently there are two "ways" to normalize vectors, matrices, and tensors.
1. `tf.math.l2_normalize`
2. `tf.norm`

The first one directly normalizes the input tensor and returns it.
The latter only computes the norm and the user has to do the division to obtain a normalized tensor.

I propose to add a function called `tf.math.normalize` which has the same/similar `args` as `tf.norm`. It computes the normalized tensor as `tf.math.l2_normalize` does but supports all the normalizations that `tf.norm` supports.

**Will this change the current api? How?**
No, it just adds a cleaner interface to do normalization.
In the long run `tf.math.l2_normalize` would be obsolete, though. It's implementation could be removed and it could use `tf.normalize` internally.

**Who will benefit with this feature?**
When looking for a way to perform l1 normalization I only found `l2_normalize` and `tf.norm`. I then expected there to exist a `l1_normalize` function as well which is not the case.
From my point of view this is an inconsistency that people will encounter from time to time.

I would propose to only add this to tensorflow 2.0 and maybe also remove `tf.math.l2_normalize` in that case.

**Comment by alextp** [comment]
Sounds reasonable. I don't think we can deprecate the existing endpoint but I'm happy to add a new one, if someone wants to contribute.

**Comment by astropeak** [comment]
Hey, I'd like to take a crack at this!

**Comment by sleighsoft** [comment]
@astropeak, I am also happy to contribute, maybe we can work together on this?

**Comment by astropeak** [comment]
@sleighsoft , in that case, maybe you could just do it. I haven't started and could always take other tasks.

**Comment by astropeak** [comment]
But if the cooperation is a better way, I am happy to work together. How to split the tasks? How we commit codes? 
Anyway, I am totally OK that only you work on the task if the cooperation isn't a good choice.

**Comment by sleighsoft** [comment]
@astropeak It looks like you have some other parts of tensorflow already in your pipeline. I would like to give it a try and would then like to get your advice once I open a PR. If you already have ideas and suggestions feel free to add them here.
My intial thought was to do it along the lines of `sklearn.preprocessing.normalize` and `numpy.linalg.norm`.

**Comment by sleighsoft** [comment]
I am, just don't have much spare time right now to make fast progress.

**Comment by jvishnuvardhan** [comment]
@sleighsoft I see your PR merged. If this issue was resolved by your PR, Please close. Thanks!
